python/ray/util/placement_group.py

```diff
@@ -34,11 +43,13 @@ class PlacementGroup:
     def empty():
         return PlacementGroup(PlacementGroupID.nil())
 
-    def __init__(self, id: PlacementGroupID):
+    def __init__(self,
+                 id: PlacementGroupID,
+                 bundle_cache: Optional[List[Dict]] = None):
         self.id = id
-        self.bundle_cache = None
+        self.bundle_cache = bundle_cache
 
-    def ready(self) -> ObjectRef:
+    def ready(self) -> Union[ObjectRef, "ClientObjectRef"]:
```

Comment:
Unfortunately, you need to give mypy a forward-reference by putting "ClientObjectRef" in quotes, because `ClientObjectRef` is behind `if TYPE_CHECKING`.
Otherwise, you get a module not found error.